client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+  const [isEditing, setIsEditing] = useState(false)
+
+  const currentSheetIndex = sheets.findIndex(
+    sheet => sheet.id === selectedTabId
+  )
+
+  // ---------- Handlers (start) ----------
+
+  const updateCurrentSheet = async (updatedSheet: IBatchResultTallySheet) => {
+    const updatedSheets = sheets.map((sheet, i) =>
+      i === currentSheetIndex ? { ...updatedSheet, id: sheet.id } : sheet
+    )
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  const addSheet = async () => {
+    const isSheetNameAlreadyTaken = (sheetName: string) =>
+      sheets.some(sheet => sheet.name === sheetName)
+
+    let incrementToAvoidNamingConflict = 0
+    let newSheetName = defaultSheetName(sheets.length + 1)
+    while (isSheetNameAlreadyTaken(newSheetName)) {
+      incrementToAvoidNamingConflict += 1
+      newSheetName = defaultSheetName(
+        sheets.length + 1 + incrementToAvoidNamingConflict
+      )
+    }
+    const newSheet = sheetWithId(constructEmptySheet(newSheetName, contest))
+    const updatedSheets = [...sheets, newSheet]
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    setSelectedTabId(newSheet.id)
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  const deleteSheet = async () => {
+    let updatedSheets = sheets.filter((_, i) => i !== currentSheetIndex)
+    if (updatedSheets.length === 1) {
+      // If we're dropping back to 1 sheet, reset the name of that sheet back to the default
+      updatedSheets = [{ ...updatedSheets[0], name: defaultSheetName(1) }]
+    }
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    setSelectedTabId(
+      // Auto-select the next sheet (or last sheet if none)
+      updatedSheets[Math.min(currentSheetIndex, updatedSheets.length - 1)].id
+    )
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  // ---------- Handlers (end) ----------
+
+  return (
+    <Detail>
+      <H4>{batch.name}</H4>
+
+      {/* When editing one of multiple sheets, BatchResultTallySheet renders its own tab bar with
+        a sheet name form input replacing the selected tab, so we hide the true tab bar */}
+      {(!isEditing || sheets.length === 1) && (
+        <Tabs
+          id={batch.name}
+          onChange={(newTabId: string) => {
+            setSelectedTabId(newTabId)
+          }}
+          selectedTabId={selectedTabId}
+        >
+          {tabs.map(tab => (
+            <Tab id={tab.id} key={tab.id}>
+              {tab.name}
+            </Tab>
+          ))}
+          <Tabs.Expander />
+          {sheets.length > 1 && (
+            <Button
+              icon="add"
+              disabled={areResultsFinalized}
+              minimal
+              onClick={addSheet}
+            >
+              Add Sheet
```

Comment:
It might be nice to go straight into edit mode when adding a sheet. Is that easy? Bonus: auto focus the tally sheet name input as well.

If this is possible, maybe we could have empty inputs instead of default to 0 tallies.

Reply:
Also, I noticed that if I click add sheet, click edit sheet, and then don't fill it out and navigate to another batch, the sheet is still saved with 0 for each tally. This makes sense in the current world, but if we automatically went into edit mode, maybe adding a sheet doesn't actually save the sheet until I enter tallies? The new sheet could be considered an unsaved change (requiring discard)

Reply:
I started with this direction when first working on this PR, but one thing that gave me pause was the fact that the "Discard Changes" button could now do one of two things, either discard edits while leaving the sheet intact (for an existing sheet) or discard the whole sheet (for a new sheet).

Maybe that's okay if we change the copy for these two cases, e.g. "Discard Sheet" vs. "Discard Changes"

Reply:
Hmm, interesting point.

Imagining we leave the copy as "Discard Changes" - if you were a user and you made a new sheet and then clicked "Discard Changes," what would you expect to happen to the new sheet? I think I would expect the new sheet to get discarded completely. So maybe it won't be that confusing in practice? I worry that changing the copy might add some cognitive overhead where you have to read the button more closely.

Reply:
Thinking some more, this makes sense! "Discard Changes" is, in both cases, just discarding everything since the last save. In one case, that's edits to an existing sheet. In another, that's the creation of a new sheet plus edits to it, if any.

https://github.com/votingworks/arlo/pull/1661/commits/3137729a933ae05003476dd5e27b70f7269381c2

^ Required some code complexity to distinguish those two cases, but that's expected.

-----

I'm still thinking through how to best switch from 0 to empty defaults in vote choice inputs. This is straightforward for all cases but one: when you haven't yet entered any tallies and click "Use Multiple Tally Sheets". We automatically take you to Sheet 2, but Sheet 1 still needs to be filled out, too. And our new model of editing/saving one sheet at a time isn't conducive to navigating errors across multiple sheets. (I'm remembering now that this case was another reason why I diverged from the "drop you off in edit mode after creating a sheet" UI that we used in the first PR.)

The simplest approach here is probably just to fill in Sheet 1 with 0s that you can then edit.

Reply:
> The simplest approach here is probably just to fill in Sheet 1 with 0s that you can then edit after saving Sheet 2.

^ Ended up going with this: https://github.com/votingworks/arlo/pull/1661/commits/929c746776da909c62c66ca2a5910048bedd8567

Reply:
In terms of auto-focusing the name input, will discuss that in the other comment about auto-focus

Reply:
Hmm, yes, that's a tricky case. I think the solution you chose works fine, but I do have another couple ideas for consideration:
- When switching to multiple tally sheets with no tally entered, drop the user into edit mode on Sheet 1 instead of Sheet 2 (that might help maintain the mental model of moving left-to-right to fill out the sheets)
- Don't auto-create a Sheet 2 at all. Switching to multiple tally sheets just enables multi-sheet mode, and then the second sheet has to be added by clicking add sheet. This might also have the benefit in the case where votes were already entered of showing the user immediately that their existing votes got transferred to Sheet 1. This would require allowing one sheet at a time to exist. So we'd also need a way to exit "multi-tally-sheet-mode" and collapse back to just showing vote totals (maybe a button in the Edit Sheets dropdown menu that's only enabled when there's only one sheet?)

Curious to hear if you think either of these directions is worth pursuing now

Reply:
> When switching to multiple tally sheets with no tally entered, drop the user into edit mode on Sheet 1 instead of Sheet 2 (that might help maintain the mental model of moving left-to-right to fill out the sheets)

I like this solution for now! I think that it also shares this benefit that you mention for the second solution:

> This might also have the benefit in the case where votes were already entered of showing the user immediately that their existing votes got transferred to Sheet 1.

The second solution is more involved and will require a backend data model update if we'd like to persist whether or not you're in multi-sheet mode, so if we were to pursue it, I'd want to later. I also think that the distinction between one sheet in single-sheet mode and one sheet in multi-sheet mode _may_ create some user confusion. I'd want to hear from us internally / real users externally whether the current model is causing confusion before investing time in this change that may or may not minimize confusion.

Reply:
https://github.com/votingworks/arlo/pull/1661/commits/b853017fc6e710f95ed7007a5e7046c59862633c

https://user-images.githubusercontent.com/12616928/199578535-244e1b97-266e-4aaf-a589-ef812fea96f2.mov